backend_new/src/utilities/custom-exception-filter.ts

```diff
@@ -0,0 +1,19 @@
+import { ArgumentsHost, Catch } from '@nestjs/common';
+import { BaseExceptionFilter } from '@nestjs/core';
+
+/*
+    This creates a simple custom catch all exception filter for us
+    As for right now its just a pass through, but as we find more errors that we don't want the 
+    front end to be exposed to this will grow
+*/
+@Catch()
+export class CustomExceptionFilter extends BaseExceptionFilter {
+  catch(exception: any, host: ArgumentsHost) {
+    console.error({
+      message: exception?.response?.message,
+      stack: exception.stack,
+      exception,
+    });
```

Comment:
I wonder if we should use the nestjs logger instead of console.error

Reply:
I went ahead and switched to using the logger, I think there are some limitations (like the logger can't be injected into the exception filter, so I created an instance for the exception handler

lemme know what you think